modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,133 @@
+        return {unlocked: true, setUnlocked: () => void 0};
+    }
+    return ctx;
+};
+
+type ConfirmationDialogProviderProps = {
+    defaultUnlocked?: boolean;
+    children?: ReactNode;
+};
+
+const ConfirmationDialogProvider = ({defaultUnlocked = true, children}: ConfirmationDialogProviderProps): ReactElement => {
+    const [unlocked, setUnlocked] = useState<boolean>(defaultUnlocked);
+
+    const value = useMemo<ConfirmationDialogContextValue>(() => ({unlocked, setUnlocked}), [unlocked]);
+
+    return <ConfirmationDialogContext.Provider value={value}>{children}</ConfirmationDialogContext.Provider>;
+};
+
+//
+// Content: wraps Dialog.Content with provider to keep shared state
+//
+
+export type ConfirmationDialogContentProps = {
+    defaultUnlocked?: boolean;
+    className?: string;
+    children?: ReactNode;
+} & ComponentPropsWithoutRef<typeof Dialog.Content>;
+
+const ConfirmationDialogContent = forwardRef<HTMLDivElement, ConfirmationDialogContentProps>(
+    ({defaultUnlocked = true, className, children, ...props}, ref): ReactElement => {
+        return (
+            <Dialog.Content ref={ref}
+                            className={cn('!outline-none [&_button]:!outline-none gap-3 min-w-170 max-w-fit', className)} {...props}>
+                <ConfirmationDialogProvider defaultUnlocked={defaultUnlocked}>{children}</ConfirmationDialogProvider>
+            </Dialog.Content>
+        );
+    }
+);
+ConfirmationDialogContent.displayName = 'ConfirmationDialog.Content';
+
+//
+// Footer: renders Cancel/Confirm actions and respects unlocked state
+//
+
+type ButtonClick = NonNullable<ComponentPropsWithoutRef<'button'>['onClick']>;
+
+export type ConfirmationDialogFooterProps = {
+    className?: string;
+    cancelLabel?: ReactNode;
+    submitLabel?: ReactNode;
+    onCancel?: ButtonClick;
+    onConfirm?: ButtonClick;
+} & ComponentPropsWithoutRef<'footer'>;
+
+const ConfirmationDialogFooter = ({
+                                      className,
+                                      cancelLabel = i18n('action.cancel'),
+                                      submitLabel = i18n('action.submit'),
+                                      onCancel,
+                                      onConfirm,
+                                      ...props
+                                  }: ConfirmationDialogFooterProps): ReactElement => {
+    const {unlocked} = useConfirmationDialog();
+
+    const handleCancel = useCallback<ButtonClick>(
+        (e) => {
+            onCancel?.(e);
+        },
+        [onCancel]
+    );
+
+    const handleSubmit = useCallback<ButtonClick>(
+        (e) => {
+            onConfirm?.(e);
+        },
+        [onConfirm]
+    );
+
+    return (
+        <Dialog.Footer {...props}>
+            <Dialog.Close asChild>
+                <Button type='button' variant='outline' onClick={handleCancel}>
+                    {cancelLabel}
+                </Button>
```

Comment:
- `type='button'` is meaningless, as it's already set in Enonic UI `<Button>`.
- Make `*Label` a `string`. Then you can pass it via `label` property instead of a child. Then it'll automatically set proper `aria-label` value too.